kyo-cats/shared/src/main/scala/kyo/Cats.scala

```diff
@@ -34,7 +34,7 @@ object Cats:
                 .map { fiber =>
                     CatsIO.async[CatsIO[A]] { cb =>
                         CatsIO {
-                            fiber.unsafe.onComplete(r => cb(r.toEither))
+                            fiber.unsafe.onResult(r => cb(r.toEither))
```

Comment:
I've renamed `IOPromise.onComplete` to `onResult` so it doesn't conflict with the `Fiber.onComplete` extension method in `Async.scala`.